File: 1.Two_Sum.go
package main

func twoSum(nums []int, target int) []int {
	var seen map[int]int
	seen = make(map[int]int)

	for i, num := range nums {
		if val, ok := seen[target-num]; ok {
			return []int{val, i}
		}
		seen[num] = i
	}

	return []int{-1, -1}
}

File: 1.Two_Sum.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        seen = {}

        for i, num in enumerate(nums):
            if target - num in seen:
                return [seen[target - num], i]
            seen[num] = i

